be/src/io/cache/block_file_cache.cpp

```diff
@@ -573,45 +572,34 @@ void BlockFileCache::recycle_deleted_blocks() {
     int i = 0;
     std::condition_variable cond;
     auto start_time = steady_clock::time_point();
-    if (_async_clear_file_cache) {
-        LOG_INFO("Start clear file cache async").tag("path", _cache_base_path);
-        auto remove_file_block = [&cache_lock, this](FileBlockCell* cell) {
-            std::lock_guard segment_lock(cell->file_block->_mutex);
-            remove(cell->file_block, cache_lock, segment_lock);
-        };
-        static int remove_batch = 100;
-        TEST_SYNC_POINT_CALLBACK("BlockFileCache::set_remove_batch", &remove_batch);
-        int i = 0;
-        std::condition_variable cond;
-        auto iter_queue = [&](LRUQueue& queue) {
-            bool end = false;
-            while (queue.get_capacity(cache_lock) != 0 && !end) {
-                std::vector<FileBlockCell*> cells;
-                for (const auto& [entry_key, entry_offset, _] : queue) {
-                    if (i == remove_batch) {
-                        i = 0;
-                        break;
-                    }
-                    auto* cell = get_cell(entry_key, entry_offset, cache_lock);
-                    if (!cell) continue;
-                    if (!cell->is_deleted) {
-                        end = true;
-                        break;
-                    } else if (cell->releasable()) {
-                        i++;
-                        cells.push_back(cell);
-                    }
+
+    LOG_INFO("Start clear file cache async").tag("path", _cache_base_path);
+    auto iter_queue = [&](LRUQueue& queue) {
+        bool end = false;
+        while (queue.get_capacity(cache_lock) != 0 && !end) {
+            std::vector<FileBlockCell*> cells;
+            for (const auto& [entry_key, entry_offset, _] : queue) {
```

Comment:
it seems that we should not iterate from the beginning again?
there may be performance penalty if there are lots of running query while we are deleting elements that are far from the head.
can we iterate from rbegin() to get better performance?